addon/components/polaris-text-field.js

```diff
@@ -561,11 +561,6 @@ export default Component.extend(
 
       let { id, type } = this.getProperties('id', 'type');
 
-      assert(
-        `ember-polaris::polaris-text-field - ${type} is not a valid type.`,
-        allowedTypes.indexOf(type) > -1
-      );
-
```

Comment:
@tomnez not sure if this was needed but it was causing a test error on the new tests I've added here so I ripped it out 😬

Reply:
Hm I guess we can leave it up to users to just use `polaris-text-field` responsibly if it's causing issues 🤷‍♂️ 

Alternatively, are there types you could add into `allowedTypes` that would stop the test from erroring out?

Reply:
I don't hugely want to have to add `undefined` to the whitelist... 😬

Reply:
Haha oh, I was thinking it might be a real value (didn't look through the whole PR yet) so yeah, lets not do that 😝